app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -90,9 +99,35 @@ class MessageActivity : AppCompatActivity() {
             }
         }
 
+        previewButtonDrawable.setOnClickListener {
+            if (drawableRecyclerAdapter.selectedPosition != -1)
+                previewBadge.setValue(Converters.convertDrawableToLEDHex(drawableRecyclerAdapter.selectedPosition?.let { it1 -> listOfDrawables?.get(it1)?.image }
+                        ?: resources.getDrawable(R.drawable.invader)) as java.util.ArrayList<String>)
+            else
+                Toast.makeText(this, getString(R.string.select_drawable), Toast.LENGTH_LONG).show()
+        }
+
+        setupRecycler()
+
         prepareForScan()
     }
 
+    var listOfDrawables: ArrayList<DrawableInfo>? = null
```

Comment:
Why creating a global variable. And what's the obsession with nullable?

Reply:
Sorry, Ill get rid of it completely

Reply:
global variable was used to fetch the element from the list as I was returning the position of the element. But I changed it to return it the dataclass instance itself. So that will get rid of the global variable